src/test/java/org/reaktivity/nukleus/tcp/internal/streams/rfc793/ClientResetAndAbortIT.java

```diff
@@ -180,5 +180,46 @@ public void shouldShutdownInputWhenClientSendsReset() throws Exception
         }
     }
 
+    @Test
+    @Specification({
+        "${route}/client/controller",
+        "${client}/client.sent.reset.and.end/client"
+    })
+    @BMRule(name = "shutdownInput",
+    targetClass = "^java.nio.channels.SocketChannel",
+    targetMethod = "shutdownInput()",
+    helper = "org.reaktivity.nukleus.tcp.internal.SocketChannelHelper$CountDownHelper",
+    condition =
+      "callerMatches(\"org.reaktivity.nukleus.tcp.internal.stream.ReadStream..*\", true, true)",
+      action = "countDown()"
+    )
+    public void shouldShutdownInputWhenClientSendsResetAndEnd() throws Exception
+    {
+        CountDownLatch shutdownInputCalled = new CountDownLatch(1);
+        CountDownHelper.initialize(shutdownInputCalled);
+
+        try (ServerSocketChannel server = ServerSocketChannel.open())
+        {
+            server.setOption(SO_REUSEADDR, true);
+            server.bind(new InetSocketAddress("127.0.0.1", 0x1f90));
 
+            k3po.start();
+            k3po.awaitBarrier("ROUTED_CLIENT");
+
+            try (SocketChannel channel = server.accept())
+            {
+                channel.configureBlocking(false);
+
+                channel.write(ByteBuffer.wrap("some data".getBytes()));
```

Comment:
Why did you not make this test contain the same code as shouldShutdownOutputAndInputWhenClientSendsAbortAndReset as I had suggested? I would expect the behavior would be the same. Did you find that didn't work?

The scripts don't need to read or write data, see my comments on the other PR (https://github.com/reaktivity/nukleus-tcp.spec/pull/20)

Reply:
I've modified it according to the rpt scripts. It failed when had the read/write data. But now it's updated.